logos/logos-workernode/logos_worker_node/model_cache.py

```
@@ -141,9 +144,117 @@ async def ensure_cached(self, model_name: str) -> str:
+        size_mb = self.model_size_bytes(model_name) / (1024 * 1024)
+        t0 = time.monotonic()
+        logger.info(
+            "Copying %s into RAM cache (%.0f MB) [sync] ...",
+            model_name, size_mb,
+        )
+
+        try:
+            rsync_available = shutil.which("rsync") is not None
+            if rsync_available:
+                result = subprocess.run(  # noqa: S603
+                    ["rsync", "-aL", "--delete", str(src) + "/", str(partial) + "/"],
```

Ruff (0.15.10)

[error] Starting a process with a partial executable path

(S607)